gym-unity/gym_unity/envs/__init__.py

```diff
@@ -121,6 +124,7 @@ def __init__(
         step_result = self._env.get_step_result(self.brain_name)
         self._check_agents(step_result.n_agents())
         self._previous_step_result = step_result
+        self._gym_id_order = list(self._previous_step_result.agent_id)
```

Comment:
Is there a way we could make _gym_id_order a dict of agent_id to index instead of a List? That way we don't have to do O(N) index operations

Reply:
Probably don't want to do this but it might work: https://stackoverflow.com/questions/1456373/two-way-reverse-map

Reply:
Unfortunately, you would have to go both ways : id to index and index to id. Having a list id implicitly a dict from index to id.

Reply:
I see so you'd have to do the two way dict to make it O(1) in both directions. If this code is only called when an agent is Done it might be OK. Could be horrific for 1000's of agents though.

Reply:
I think it's worth trying to remove these index calls if possible. But getting tests first is more important, then you can optimize.

Reply:
I added 2 tests for sanitize info